packages/ethereum/cardstack/service.js

```diff
@@ -24,13 +25,13 @@ class EthereumService {
     this._indexer = indexer;
     this._providers = new WeakMap();
     this._eventListeners = new WeakMap();
-    this._hasStartedListening = false;
+    this._hasStartedListening = {};
     this._hasConnected = false;
-    this._isIndexing = false;
-    this._indexQueue = [];
+    this._isIndexing = {};
+    this._indexQueue = {};
     this._processQueueTimeoutMs = 1000;
-    this._processQueueTimeout = null;
-    this._contractDefinitions = null;
+    this._processQueueTimeout = new WeakMap();
+    this._contractDefinitions = new WeakMap();
```

Comment:
I think the keys you're using in both these WeakMaps are strings. If that's so, they're probably not doing what you want them to -- they will act like plain old Maps and the values will be retained forever.

Reply:
whoops--yes. will fix.